ci-operator/step-registry/baremetal/lab/post/add-nodes/baremetal-lab-post-add-nodes-commands.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+set -o nounset
+set -o pipefail
+
+CLUSTER_NAME="${NAMESPACE}"
+DIR=/tmp/bmo
+mkdir -p ${DIR}
+
+master00_addr=$(yq e '.[] | select(.name == "master-00") | .bmc_address' ${SHARED_DIR}/hosts.yaml)
+master00_bootMacAddr=$(yq e '.[] | select(.name == "master-00") | .provisioning_mac' ${SHARED_DIR}/hosts.yaml)
+master00_username=$(yq e '.[] | select(.name == "master-00") | .bmc_user' ${SHARED_DIR}/hosts.yaml)
+master00_pass=$(yq e '.[] | select(.name == "master-00") | .bmc_pass' ${SHARED_DIR}/hosts.yaml)
+
+workera_addr=$(yq e '.[] | select(.name == "worker-a-00") | .bmc_address' ${SHARED_DIR}/hosts.yaml)
+workera_bootMacAddr=$(yq e '.[] | select(.name == "worker-a-00") | .provisioning_mac' ${SHARED_DIR}/hosts.yaml)
+workera_username=$(yq e '.[] | select(.name == "worker-a-00") | .bmc_user' ${SHARED_DIR}/hosts.yaml)
+workera_pass=$(yq e '.[] | select(.name == "worker-a-00") | .bmc_pass' ${SHARED_DIR}/hosts.yaml)
+
+cat > ${DIR}/bmo.yaml <<EOF
+apiVersion: metal3.io/v1alpha1
+kind: Provisioning
+metadata:
+  name: provisioning-configuration
+spec:
+  provisioningNetwork: "Disabled"
+  watchAllNamespaces: false
+
+EOF
+
+cat > ${DIR}/master00-bmc.yaml <<EOF
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: master00-bmc
+  namespace: openshift-machine-api
+type: Opaque
+data:
+  username: "$master00_username"
+  password: "$master00_pass"
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: master00
+  namespace: openshift-machine-api
+spec:
+  bmc:
+    address: "redfish-virtualmedia://$master00_addr/redfish/v1/Systems/1"
+    credentialsName: "master00-bmc"
+    disableCertificateVerification: True
+  bootMACAddress: "$master00_bootMacAddr"
+  customDeploy:
+    method: install_coreos
+  externallyProvisioned: true
+  online: true
+  userData:
+    name: master-user-data-managed
+    namespace: openshift-machine-api
+
+EOF
+
+cat > ${DIR}/workera-bmc.yaml <<EOF
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: workera-bmc
+  namespace: openshift-machine-api
+type: Opaque
+data:
+  username: "$workera_username"
+  password: "$worker_pass"
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: worker-a-00
+  namespace: openshift-machine-api
+spec:
+  bmc:
+    address: "redfish-virtualmedia://$workera_addr/redfish/v1/Systems/1"
```

Comment:
we have a variable within the inventory to get the scheme and the base URL to use. Respectively, `bmc_scheme` and `bmc_base_uri`

Reply:
Where to use these variables? I have used `name` `bmc_user` `bmc_pass` and `bmc_address`.
Are the username and password variables used correct? or do I need to use `redfish_*` variables?

Reply:
right, for the redfish credentials, it's better to use redfish_user and redfish_password (rationale: some nodes have different credentials for IPMI and redfish and they are kind of hard-coded :( )